### ISSUE TITLE
Got last emmited state when navigate to same screen which was visited before

### ISSUE THREAD

**Issue by dev-devarsh4** [felangel/bloc#3967]
**Description**
I have a Home screen widget with a button in the center, on tap of a button I navigate to the search screen  In which I have a text field with a Search and list to get search result

I have created a search cubit with blocProvider in the MyApp widget

**Steps To Reproduce**

In the search screen if I search then with help of onChanged of textfield I call search people method with searchCubit and the search state is emitted by cubit
But if I go back to the home screen and navigate to the Search screen again I get a search list that was searched by the previous as mentioned before

In short, I got the last emitted state in the particular screen that I have visited earlier

**Expected Behavior**
I should not get the Last state emitted, instead, I got InitalState which is generated by boilerplate code

**Comment by felangel** [comment]
Hi that likely means you’re not scoping the bloc low enough in the widget tree. The cubit should be provided in the search page and I’m guessing you’re providing it higher in the widget tree which is why it never gets disposed. Hope that helps 👍

**Comment by dev-devarsh4** [comment]
It was just an example I have so many search screens in my project If I provide a Search Cubit to every search screen, Every time that search screen is built, a new instance of Search cubit will be created and it is not performance-orientated practice
What I need if visit the search screen again the default value should be return of the bloc builder
As below
![Screenshot 2023-10-06 at 10 10 42 AM](https://github.com/felangel/bloc/assets/126959872/3c3385f3-8d61-4cd5-9a8e-f195319a2e37)

For Example

**Comment by felangel** [comment]
You either need to scope the bloc lower in the widget tree as I mentioned (this should actually improve performance since the bloc will be disposed when the search widget is unmounted) or you need to manually add some sort of "reset" event. Hope that helps 👍
